src/raftstore/store/peer.rs

```diff
@@ -152,6 +153,12 @@ impl PendingCmdQueue {
     }
 }
 
+pub enum RequestType {
+    LeaseRead,
+    FollowerRead,
```

Comment:
IMO, these are three types of read/write in total:
1. lease read, only happens on leader with valid lease. It doesn't write to raft log.
2. read index read, could happens on follower or leader. No write to raft log.
3. consistent read/write, only happens on leader. It writes to raft log.

The name `FollowerRead` indicates that the read index read only happens on follower. That is not true. Because the code of this PR supports read index read on a leader.
And `Others` is somewhat casual. It could be given a better name with clear meaning.